frontend/src/metabase/components/Schedule/AutoWidthSelect.tsx

```diff
@@ -0,0 +1,39 @@
+import { useMemo } from "react";
+
+import { useSelector } from "metabase/lib/redux";
+import { getSetting } from "metabase/selectors/settings";
+import type { SelectProps } from "metabase/ui";
+import { Select } from "metabase/ui";
+import type { FontStyle } from "metabase/visualizations/shared/types/measure-text";
+
+import { getLongestSelectLabel, measureTextWidthSafely } from "./utils";
+
+/** A Select that is automatically sized to fit its largest option name */
+export const AutoWidthSelect = <Value extends string>({
+  style,
+  value,
+  ...props
+}: { style?: Partial<FontStyle>; value: Value } & SelectProps<Value>) => {
+  const fontFamily = useSelector(state =>
```

Comment:
I think it you move this logic outside, this component would be quite generic and maybe it could go to `ui/components/inputs`?

Reply:
This is a good idea! I think we could do this as a follow up PR so you're not blocked for much longer